Abyss helper

dvrooman opened the pull request:
### Updates to Abyss Helper
- made new setup  section:
	- tags first
	- splits Abyss into two decks: top and bottom
	- players add cards/investigators to top half
	- shuffles and merges two halves back into Abyss
- changed tags to AbyssMatColor instead of MatColor _(easier to find in Tag window until we have a better interface)_
- updates to vscode tasks/scripts to work with macOS as well as windows

Chr1Z93 commented on library.json:
seems like this shouldn't be in this PR

dvrooman commented on .vscode/decompose_saved_object.py:
```diff
@@ -81,15 +81,27 @@ def decompose_saved_object(input_file):
-        f"{nickname}.json",
-    )
+    if sys.platform == "linux" or sys.platform == "linux2": # linux
+        pass
```
whoops. forgot to get rid of that. left over from code I was referencing to figure out multiple platforms…

dvrooman commented on .vscode/build_saved_object.py:
```diff
@@ -21,31 +21,58 @@ def build_saved_object(input_file):
-    ]
+    if sys.platform == "darwin": # macOS
+        cmd = [
+            "TTSModManager", # compiled ModManager
```
You're probably right: I'll see if I can get my builds working with the pure source.

dvrooman commented on .vscode/tasks.json:
```diff
@@ -5,10 +5,18 @@
+      },
+      "osx":{
+        "args": [
+          "${workspaceFolder}/.vscode/build_saved_object.py",
```
I'll double check this, but I'm pretty sure the backslash path type for windows doesn't work on macOS. I currently don't have a way to test on Windows, so I have no idea if the forward slash works there.

dvrooman commented on decomposed/campaign/Return to The Path to Carcosa/ReturntoThePathtoCarcosa.e9889a/2TheLastKing.6730a1.luascriptstate:
I'm pretty sure this and the 6 other files came from the "Merge branch 'main' into abyss-helper" commit. I was trying to make sure my branch was up to date with upstream/main and I believe I used "Update from upstream/main" on my branch in the Github Desktop. In this case, the files have no bearing on my branch. In the future, though, how should I go about keeping my branch up to date? I've tried a bunch of tests and looked around online, but I'm getting more and more confused.

For this PR, I'm not sure how to proceed. I think doing "Revert Changes in Commit" on the "Merge branch 'main' into abyss-helper" commit and then pushing is the way forward?

dvrooman commented:
I believe I've addressed the comments so far. I don't currently have any way to test everything in Windows. I did my best to not touch code there. To get decompose working on macOS, I did have to change something in TTSModManager. I can open a PR over there if you think it's a good idea.

dvrooman commented:
reverted commits with changes for cross-platform compiling. moved those to a separate PR.
